### PR TITLE
Document training settings

### PR DESCRIPTION
**Pull request by guarin**
## What has changed and why?

* Add settings page to docs
* Simplify task specific docs pages
* Add missing default task transform values

For review: Build the docs and check the settings page and task specific docs pages :)

## How has it been tested?

[Train Settings - LightlyTrain documentation.pdf](https://github.com/user-attachments/files/24770883/Train.Settings.-.LightlyTrain.documentation.pdf)

## Did you update [CHANGELOG.md](../CHANGELOG.md)?

- [ ] Yes
- [x] Not needed (internal change)

## Did you update the documentation?

- [x] Yes
- [ ] Not needed (internal change without effects for user)

### CODE REVIEW

File: docs/source/settings/train_settings.md [hunk 1]
```diff
@@ -0,0 +1,765 @@
...
+| [`model_args`](#model_args)                     | `dict`                        | `None`         | Task/model-specific training hyperparameters.                                                                                                                       |
+| [`steps`](#steps)                               | `int`                         | `"auto"`       | Number of training steps. `"auto"` selects a model-dependent default.                                                                                               |
+| [`precision`](#precision)                       | `str`                         | `"bf16-mixed"` | Numeric precision mode (e.g. `"16-true"`, `"32-true"`, `"bf16-mixed"`).                                                                                             |
+| [`batch_size`](#batch_size)                     | `int`                         | `"auto"`       | Global batch size across all devices. Per-device batch size is derived from `batch_size / (devices * num_nodes)`.                                                   |
```
Comment (stegmuel) [inline docs/source/settings/train_settings.md]: With the below description of `devices`, the formula can be misleading.

File: docs/source/settings/train_settings.md [hunk 2]
```diff
@@ -0,0 +1,765 @@
...
+listed on this page.
+```
+
+(train-settings-output)=
+
+## Output
+
+### `out`
+
+The output directory where the model checkpoints and logs are saved. We recommend to
```
Comment (stegmuel) [inline docs/source/settings/train_settings.md]: This is even a requirement no? The training crashes if the same `out` and `data` are use for concurrent runs. Maybe this could be explained.

File: docs/source/settings/train_settings.md [hunk 3]
```diff
@@ -0,0 +1,765 @@
...
+
+- [Object Detection](object-detection-data)
+- [Instance Segmentation](instance-segmentation-data)
+- [Panoptic Segmentation](panoptic-segmentation-data)
+- [Semantic Segmentation](semantic-segmentation-data)
+
+### `batch_size`
+
+Global batch size across all devices. The batch size per device/GPU is computed as
+`batch_size / (devices * num_nodes)`. By default, `batch_size` is set to `"auto"`, which
```
Comment (stegmuel) [inline docs/source/settings/train_settings.md]: Same remark as above

File: docs/source/settings/train_settings.md [hunk 4]
```diff
@@ -0,0 +1,765 @@
...
+
+### `model_args`
+
+Dictionary with model-specific training parameters. The available keys vary by
+architecture. The table lists the most commonly tuned options:
+
+| Key                                             | Type                      | Description                        |
+| ----------------------------------------------- | ------------------------- | ---------------------------------- |
+| [`lr`](#lr)                                     | `float`                   | Base learning rate.                |
+| [`backbone_weights`](#backbone_weights)         | `Path`<br>`str`<br>`None` | Path to backbone weights to load.  |
```
Comment (stegmuel) [inline docs/source/settings/train_settings.md:106]: Should something be said about the fact that backbone weights will be overwritter when full model weights are provided?

File: docs/source/settings/train_settings.md [hunk 5]
```diff
@@ -0,0 +1,765 @@
...
+value. Reduce for shorter training or increase to train longer.
+
+Epoch based training is currently not supported.
+
+### `precision`
+
+Training precision setting. Must be one of the following strings:
+
+- `"bf16-mixed"`: Default. Operations run in bfloat16 where supported, weights are saved
+  in float32.
```
Comment (stegmuel) [inline docs/source/settings/train_settings.md]: This is also not supported on all hardware, no?

File: docs/source/settings/train_settings.md [hunk 6]
```diff
@@ -0,0 +1,765 @@
...
+   - This initializes model weights from the checkpoint but starts a fresh training
+     state.
+
+### `resume_interrupted`
+
+Use when a run terminates unexpectedly and you want to continue from the latest
+checkpoint stored in `out/checkpoints/last.ckpt`. Do not modify any other training
+arguments! This will restore the exact training state, including optimizer parameters,
+current step, and any learning rate or other schedules from the previous run. The flag
+is inteded for crash recovery only. See [](#load-checkpoint-for-a-new-run) for
```
[the remark this answers is not in the document]
Comment (stegmuel) [reply docs/source/settings/train_settings.md]: Shouldn't there be something in []?

File: docs/source/settings/train_settings.md [hunk 7]
```diff
@@ -0,0 +1,765 @@
...
+lightly_train.train_object_detection(
+    out="out/my_new_experiment",  # New output directory for new run.
+    model="out/my_experiment/exported_models/exported_best.pt",  # Load model from previous run.
+    steps=2000,  # Change training parameters as needed.
+)
+```
+
+(train-settings-checkpoint-saving)=
+
+## Checkpoint Saving
```
Comment (stegmuel) [inline docs/source/settings/train_settings.md]: I would move this section up

File: docs/source/settings/train_settings.md [hunk 8]
```diff
@@ -0,0 +1,765 @@
...
+
+LightlyTrain saves two types of checkpoints during training:
+
+1. `out/my_experiment/checkpoints`: Full checkpoints including optimizer, scheduler, and
+   training state. Used to resume training with [](#resume_interrupted).
```
Comment (stegmuel) [inline docs/source/settings/train_settings.md]: empty []

File: docs/source/settings/train_settings.md [hunk 9]
```diff
@@ -0,0 +1,765 @@
...
+    save_checkpoint_args={
+        "watch_metric": "val_metric/map",  # Use mAP as the best-checkpoint metric.
+        "mode": "max", # Higher is better for mAP. Set to "min" for metrics where lower is better.
+    },
+)
+```
+
+#### `mode`
+
+Direction used when ranking the `watch_metric` for best-checkpoint selection. Must be
```
Comment (stegmuel) [inline docs/source/settings/train_settings.md]: Operation used when comparing the ...?

File: docs/source/settings/train_settings.md [hunk 10]
```diff
@@ -0,0 +1,765 @@
...
+Check the respective task pages for the default transforms applied:
+
+- [Object Detection](object-detection-transform-args)
+- [Instance Segmentation](instance-segmentation-transform-args)
+- [Panoptic Segmentation](panoptic-segmentation-transform-args)
+- [Semantic Segmentation](semantic-segmentation-transform-args)
+
+#### `image_size`
+
+Tuple specifying the height and width of input images after cropping and resizing. The
```
Comment (stegmuel) [inline docs/source/settings/train_settings.md:636]: Should something be said about not all resolution being supported by some models?

File: docs/source/instance_segmentation.md [hunk 11]
```diff
@@ -285,128 +285,27 @@ All models are
...
 and fine-tuned by Lightly.
 
-(instance-segmentation-logging)=
-
-## Logging
+## Training Settings
 
-Logging is configured with the `logger_args` argument. The following loggers are
-supported:
+See [](train-settings) on how to configure training settings.
```
Comment (stegmuel) [inline docs/source/instance_segmentation.md:290]: [] ?

File: docs/source/instance_segmentation.md [hunk 12]
```diff
@@ -285,128 +285,27 @@ All models are
...
-        logger_args={
-            "wandb": {
-                "project": "my_project",
-                "name": "my_experiment",
-                "log_model": False,        # Set to True to upload model checkpoints
-            },
-        },
-    )
-```
+See [](train-settings-logging) on how to configure logging.
```
Comment (stegmuel) [inline docs/source/instance_segmentation.md:302]: same. Maybe this is rendered differently in the doc?

File: docs/source/settings/train_settings.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (guarin) [reply docs/source/settings/train_settings.md]: No, if you don't put anything it will automatically insert the title from the section you are referring to. However, it doesn't automatically insert inline code. So if the section title is:
```
### `devices` 
```

You'll still have to link it as:
```
[`devices`](#devices)
```

for the link to show up as inline code.

File: docs/source/settings/train_settings.md (hunk 4 above)
Comment (stegmuel) [reply docs/source/settings/train_settings.md:106]: did you write something about that?

Comment (guarin) [reply docs/source/settings/train_settings.md:106]: Yes
